### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by compnerd**

### CODE REVIEW

File: src/internal.h [hunk 1]
```diff
@@ -356,14 +356,10 @@ DISPATCH_EXPORT DISPATCH_NOTHROW void dispatch_atfork_child(void);
 		do { (elm)->field.tqe_prev = NULL; } while (0)
 
 #if DISPATCH_DEBUG
-// sys/queue.h debugging
-#if defined(__linux__)
-#define QUEUE_MACRO_DEBUG 1
```
Comment (MadCoder) [inline src/internal.h]: this breaks people using libbsd-dev and doesn't hurt, please do not remove.
what you probably want to do is to both set QUEUE_MACRO_DEBUG and TRASHIT to support both (?)

File: src/shims/getprogname.h [hunk 2]
```diff
@@ -24,9 +24,9 @@
 
 #if !HAVE_GETPROGNAME
 
-#ifdef __ANDROID__
+#if defined(__ANDROID__)
```
Comment (MadCoder) [inline src/shims/getprogname.h]: please keep the `#ifdef` only the removal of the `)` below is needed

File: src/shims/linux_stubs.h [hunk 3]
```diff
@@ -16,6 +16,16 @@
 #ifndef __DISPATCH__STUBS__INTERNAL
 #define __DISPATCH__STUBS__INTERNAL
 
+#if !defined(TAILQ_FOREACH_SAFE)
```
Comment (MadCoder) [inline src/shims/linux_stubs.h]: please keep consistent style and use `#ifndef`

File: src/shims/linux_stubs.h [hunk 4]
```diff
@@ -16,6 +16,16 @@
...
+#define TAILQ_FOREACH_SAFE(var, head, field, temp)                             \
+  for ((var) = TAILQ_FIRST((head));                                            \
+       (var) && ((temp) = TAILQ_NEXT((var), field), 1); (var) = (temp))
+#endif
+
+#if !defined(TRASHIT)
```
Comment (MadCoder) [inline src/shims/linux_stubs.h]: please don't do that, TRASHIT is only meant to be set in debug mode

File: src/shims/linux_stubs.h [hunk 5]
```diff
@@ -16,6 +16,16 @@
 #ifndef __DISPATCH__STUBS__INTERNAL
 #define __DISPATCH__STUBS__INTERNAL
 
+#if !defined(TAILQ_FOREACH_SAFE)
+#define TAILQ_FOREACH_SAFE(var, head, field, temp)                             \
+  for ((var) = TAILQ_FIRST((head));                                            \
```
Comment (MadCoder) [inline src/shims/linux_stubs.h]: please use tabs for alignment

File: src/shims/linux_stubs.h (hunk 4 above)
Comment (compnerd) [inline src/shims/linux_stubs.h]: Just following the android pattern, where the `android_stubs.h` always define it.

**Comment by MadCoder** [comment]
@compnerd :

> Just following the android pattern, where the android_stubs.h always define it.

I got the message through mail but couldn't find where it was made after your force push.
I didn't realize Android was doing that and it's wrong, TRASHIT and other similar macros are about poisoning the `<sys/queue.h>` macros, and does have a performance penalty cost for some things that are quite in hotpaths where you don't want it unless you debug. The Android port is wrong and ought to be fixed, would you mind doing it on the same pull request? the rest looks good

**Comment by compnerd** [comment]
@MadCoder sure, I can squash it into the same change.

**Comment by compnerd** [comment]
Gentle post-holiday ping for @MadCoder

File: src/shims/linux_stubs.h (hunk 3 above)
Comment (MadCoder) [inline src/shims/linux_stubs.h]: you haven't handled this review feedback.

Please use `#ifndef` and squash in a single commit before merge, thanks

**Comment by compnerd** [comment]
Address the last bit of the review comments that I had missed.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
Hmm, seems that I dont have swift-ci rights to this project.  Mind running this through @MadCoder?  BTW, whats the darwin-pending tag?

**Comment by MadCoder** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
It seems that the failure is unrelated?

**Pull request merged**